### DISCUSSION TITLE
How to pass process.env variable?

### DISCUSSION THREAD

**Discussion by PollySt** [test-results-reporter/reporter#228] [Q&A]
Hello!
I've looked into your documentation of environment variables, and as I understood, I can pass them like this `"url": "{TEAMS_URL}"` in the config file, but it doesn't work for me.
This is my config file: 
```json
{
  "targets": [
    {
      "name": "slack",
      "inputs": {
        "url": "{SLACK_TESTS_STAGING_URL}",
        "publish": "test-summary",
        "title": "Test Results - Mobile (Device: {DEVICE})"
      },
      "extensions": [
        {
          "name": "ci-info",
          "inputs": {
            "show_repository": false
          }
        },
        {
          "name": "quick-chart-test-summary"
        }
      ]
    }
  ],
  "results": [
    {
      "type": "junit",
      "files": [
        "junit-results-staging.xml"
      ]
    }
  ]
}
```
I assign a value to the variable during the test run: `process.env.DEVICE = String(randomDevice)`
But in the report, I get this:
<img width="161" alt="image" src="https://github.com/user-attachments/assets/8cd8372c-9a38-42a3-ad7b-335d26a4aed6">

Using `"title": "Test Results - Mobile (Device: {process.env.DEVICE})"` also gives a string `Mobile (Device: process.env.DEVICE)`.

Can you help me please? What am I doing wrong?
Thanks!
@ASaiAnudeep

**Comment by ASaiAnudeep** [comment]
Try storing the **DEVICE** environment variable in a `.env` file for better accessibility across steps.

```js
require('fs').writeFileSync('.env', `DEVICE=${randomDevice}`);
```

**Reply by PollySt** [reply]
With writeFileSync, the whole .env file will be overwritten. With appendFileSync a lot of DEVICE variables will be added each time. Is there a way to avoid this?

**Reply by ASaiAnudeep** [reply]
I assume, if you are running in CI, it will be appended only once. If not, may be you can read the entire file and replace the appropriate device name in the `.env` file.
